Template10 (Library)/Controls/HamburgerMenu.xaml.cs

```diff
@@ -513,64 +515,81 @@ public void UpdatePaneMarginToShowHamburgerButton()
             }
 
             // signal previous
-            if (previous != null && previous != value && previous.IsChecked.Value)
+            if (previous != null && previous != current && previous.IsChecked.Value)
             {
-                previous.IsChecked = false;
-                previous.RaiseUnselected();
-
                 // Workaround for visual state of ToggleButton not reset correctly
-                if (value != null)
+                if (current != null)
                 {
-                    var control = LoadedNavButtons.First(x => x.HamburgerButtonInfo == value).GetElement<Control>();
+                    var control = LoadedNavButtons.First(x => x.HamburgerButtonInfo == current).GetElement<Control>();
                     VisualStateManager.GoToState(control, "Normal", true);
                 }
             }
 
             // navigate only when all navigation buttons have been loaded
-            if (AllNavButtonsAreLoaded && value?.PageType != null)
+            if (AllNavButtonsAreLoaded && current?.PageType != null)
             {
-                if (await NavigationService.NavigateAsync(value.PageType, value?.PageParameter, value?.NavigationTransitionInfo))
+                if (await NavigationService.NavigateAsync(current.PageType, current?.PageParameter, current?.NavigationTransitionInfo))
                 {
+                    SignalPreviousPage(previous, current);
+                    if (current != null) SignalCurrentPage(previous, current);
+
                     IsOpen = (DisplayMode == SplitViewDisplayMode.CompactInline && IsOpen);
-                    if (value.ClearHistory)
+                    if (current.ClearHistory)
                         NavigationService.ClearHistory();
-                    if (value.ClearCache)
+                    if (current.ClearCache)
                         NavigationService.ClearCache(true);
                 }
-                else if (NavigationService.CurrentPageType == value.PageType
-                     && (NavigationService.CurrentPageParam ?? string.Empty) == (value.PageParameter ?? string.Empty))
+                else if (NavigationService.CurrentPageType == current.PageType && (NavigationService.CurrentPageParam ?? string.Empty) == (current.PageParameter ?? string.Empty))
                 {
-                    if (value.ClearHistory)
+                    SignalPreviousPage(previous, current);
+                    if (current != null) SignalCurrentPage(previous, current);
```

Comment:
Don't you think (current != null) belongs in SignalCurrentPage() instead of here?